src/Microsoft.Health.Fhir.CosmosDb/Features/Search/Queries/ExpressionQueryBuilder.cs

```diff
@@ -209,6 +217,13 @@ public object VisitMissingField(MissingFieldExpression expression, Context conte
             return null;
         }
 
+        public object VisitNotExpression(NotExpression expression, Context context)
+        {
+            string message = $"Not expression should be handled in {nameof(VisitSearchParameter)}";
+            Debug.Fail(message);
+            throw new InvalidOperationException(message);
```

Comment:
To address @johnstairs comment, can we change it to:
```
            _queryBuilder.Append("NOT (");
            expression.Expression.AcceptVisitor(this, context);
            _queryBuilder.Append(")");
            return null;
```

I've test it locally on my cosmos db and it seems working fine.